android/src/main/kotlin/com/github/fluttercommunity/flutterdownloader/DownloadWorker.kt

```diff
@@ -0,0 +1,187 @@
+package com.github.fluttercommunity.flutterdownloader
+
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.*
+import java.io.BufferedInputStream
+import java.io.FileOutputStream
+import java.net.HttpURLConnection
+import java.net.SocketException
+import java.net.URL
+import java.net.URLDecoder
+import java.util.*
+import java.util.regex.Pattern
+import kotlin.coroutines.cancellation.CancellationException
+
+/***
+ * A simple worker that will post your input back to your Flutter application.
+ *
+ * It will block the background thread until a value of either true or false is received back from Flutter code.
+ */
+class DownloadWorker(
+    applicationContext: Context,
+    workerParams: WorkerParameters
+) : CoroutineWorker(applicationContext, workerParams) {
+    companion object {
+        const val TAG = "DownloadWorker"
+    }
+/*
+    private val charsetPattern = Pattern.compile("(?i)\\bcharset=\\s*\"?([^\\s;\"]*)")
+    private val filenameStarPattern = Pattern.compile("(?i)\\bfilename\\*=([^']+)'([^']*)'\"?([^\"]+)\"?")
+    private val filenamePattern = Pattern.compile("(?i)\\bfilename=\"?([^\"]+)\"?")
+
+    private fun getCharsetFromContentType(contentType: String?): String? {
+        if (contentType == null) return null
+        val m = charsetPattern.matcher(contentType)
+        return if (m.find()) {
+            m.group(1)?.trim { it <= ' ' }?.uppercase(Locale.US)
+        } else null
+    }
+
+    private fun getFileNameFromContentDisposition(
+        disposition: String?,
+        contentCharset: String?
+    ): String? {
+        if (disposition == null) return null
+        var name: String? = null
+        var charset = contentCharset
+
+        //first, match plain filename, and then replace it with star filename, to follow the spec
+        val plainMatcher = filenamePattern.matcher(disposition)
+        if (plainMatcher.find()) name = plainMatcher.group(1)
+        val starMatcher = filenameStarPattern.matcher(disposition)
+        if (starMatcher.find()) {
+            name = starMatcher.group(3)
+            charset = starMatcher.group(1)?.uppercase(Locale.US)
+        }
+        return if (name == null) null else URLDecoder.decode(
+            name,
+            charset ?: "ISO-8859-1"
+        )
+    }
+*/
+    override suspend fun doWork(): Result {
```

Comment:
I know this is WIP, but just a heads up that it'd be good to split this method and/or add a small doc comment.

Reply:
As said I'll make a documentation pass later :)

Reply:
I hope that is fine now